### DISCUSSION TITLE
Erro ao inserir o endereço no Git do Github, o que fazer?

### DISCUSSION THREAD

**Discussion by andrejs400** [frontendbr/forum#1924] [Perguntas e Respostas] [answered]
Boa tarde pessoal.

Estou fazendo um curso inicial de Git e Github, porém na hora de eu inserir o endereço do repositório do github no git da a seguinte mensagem:

$ git push origin master
error: src refspec master does not match any
error: failed to push some refs to 'https://github.com/andrejs400/livro-receitas.git'

o que eu devo fazer? 

Obrigado pela ajuda!

**Comment by renedocket** [comment]
Fala @andrejs400 , beleza? Depois que você criou o repositório você seguiu as instruções do github?

Algo como esse: 

…or create a new repository on the command line
echo "# teste" >> README.md
git init
git add README.md
git commit -m "first commit"
git branch -M main
git remote add origin git@github.com:andrejs400/livro-receitas.git
git push -u origin main
                
…or push an existing repository from the command line
git remote add origin git@github.com:andrejs400/livro-receitas.git
git branch -M main
git push -u origin main

**Reply by BrunoJCosta** [reply]
resolveu o meu problema que estou tendo a dias vlw  :heart_eyes: :heart_eyes: :heart_eyes:

**Reply by Alexsandro-Santos-Junior** [reply]
Muito obrigado o meu estava dando erro e ao fazer essa "receita" deu certo muito obrigado!

**Comment by andrejs400** [comment, marked as answer]
Opa @renedocket , me perdoe, mas estou um pouco perdido, pois estou iniciando hj... Olha como está 👍 

$ ls
READM.md  receitas/

André@DESKTOP-GH8F576 MINGW64 /c/workspace/livro-receitas (master)
$ git remote add origin https://github.com/andrejs400/livro-receitas.git

André@DESKTOP-GH8F576 MINGW64 /c/workspace/livro-receitas (master)
$ git remote -v
origin  https://github.com/andrejs400/livro-receitas.git (fetch)
origin  https://github.com/andrejs400/livro-receitas.git (push)

André@DESKTOP-GH8F576 MINGW64 /c/workspace/livro-receitas (master)
$ git status
On branch master

No commits yet

Changes to be committed:
  (use "git rm --cached <file>..." to unstage)
        new file:   READM.md
        new file:   receitas/stogonoff.md.txt


André@DESKTOP-GH8F576 MINGW64 /c/workspace/livro-receitas (master)
$ git push origin master
error: src refspec master does not match any
error: failed to push some refs to 'https://github.com/andrejs400/livro-receitas.git'

André@DESKTOP-GH8F576 MINGW64 /c/workspace/livro-receitas (master)
$ git push origin master
error: src refspec master does not match any
error: failed to push some refs to 'https://github.com/andrejs400/livro-receitas.git'

André@DESKTOP-GH8F576 MINGW64 /c/workspace/livro-receitas (master)
$ git remote add origin https://github.com/andrejs400/livro-receita.git
fatal: remote origin already exists.

André@DESKTOP-GH8F576 MINGW64 /c/workspace/livro-receitas (master)
$ git push origin master
error: src refspec master does not match any
error: failed to push some refs to 'https://github.com/andrejs400/livro-receitas.git'

André@DESKTOP-GH8F576 MINGW64 /c/workspace/livro-receitas (master)
$

**Reply by mleilane** [reply]
OI Andre, deu certo? Estou com esse mesmo problema

**Reply by DeusSupremo** [reply]
tambem estou...

**Reply by ojvribeiro** [reply]
Acredito que você precise fazer `commit` antes de fazer o `push`.

```bash
$ git commit -m "First commit"

$ git push origin master
```

Só lembrando que o GitHub mudou recentemente o nome da branch principal de `master` para `main`. Então é preciso sempre checar direitinho o nome da sua branch.

**Reply by nycolexavier** [reply]
Obrigada, Ojvribeiro!

**Reply by ThiagoPapapietro** [reply]
Obrigado @ojvribeiro 👍🏼👍🏼👍🏼

**Reply by edvaldon** [reply]
Resolveu obrigado  o ojvribeiro

**Reply by kelibraga** [reply]
obrigado @ojvribeiro e toda a galera. Graças ao babte papo de vocês consegui resolver meu erro ! Além da branch estar errada eu executei o comando com root, e a chave SSH não estva usando esse usuário. Isso também contribuiu para dar o erro : Permission denied (publickey). fatal: Could not read from remote repository.
 Please make sure you have the correct access rights and the repository exists.

**Reply by Medeiros-Joao** [reply]
@ojvribeiro muito bom, obrigado pela ajuda, os arquinos estavam apenas em staged, por isso o erro.

**Reply by MarcelFinavaro** [reply]
Se alguém puder me ajudar agradeço.

Estou tentando executar o comando push, porém aparece esse  erro "fatal protocol "https" is not supported" como resolver

**Comment by ninetails** [comment]
Opa André, tenta rodar

> git push --set-upstream origin master

**Comment by mstuttgart** [comment]
@andrejs400 pelo que vi (posso estar enganado), parece que a sua branch ainda não possui nenhum commit feito. Isso deve estar causando o problema na hora do push. Os comandos abaixo irão adicionar os arquivos pra você poder fazer o 'push'

* git add READM.md receitas/
* git commit -m "Initial commit"
* git push origin master

**Comment by fecds** [comment]
@andrejs400 , acho que vc cometeu o mesmo erro que eu. Quando você pegou o comando abaixo no git e executou ele dá esse erro de referência.
git git branch -M main
git push -u origin main

Vc precisa executar o comando com o nome MASTER, ele vai deixar de dar o erro de referência, mas vai continuar sem fazer nada.  Estou com o mesmo problema sem conseguir subir o repositorio no github
git git branch -M master
git push -u origin master

**Reply by ojvribeiro** [reply]
Na verdade, é porque não foi feito o `commit` mesmo. Isso é visto nesse [trecho que ele postou](https://github.com/frontendbr/forum/discussions/1924#discussioncomment-162046) ali em cima:

> ```bash
> $ git status
> On branch master
> 
> No commits yet
> 
> Changes to be committed:
> (use "git rm --cached ..." to unstage)
> new file: READM.md
> new file: receitas/stogonoff.md.txt
> ```

Aqui mostra que há alterações desses dois arquivos em "stage".

E ele fez `push` na `master` mesmo:

> ```bash
> $ git push origin master
> error: src refspec master does not match any
> error: failed to push some refs to 'https://github.com/andrejs400/livro-receitas.git'
> ```

... mas é sempre bom mesmo checar se não é a `main`.


A provável solução é fazer primeiro o `commit` e então o `push`.
```bash
$ git commit -m "First commit"

$ git push origin master
```

**Reply by fecds** [reply]
Victor, pode está faltando o commit, mas acho que tem haver com o que vc falou sobre a branch. Em setembro do ano passado o github mudou o nome da branch default de MASTER para MAIN. No exercício ensina a criar a branch local como master e no github cria como main. 

O erro mais comum é quando tentar subir através do comando 

$ git push origin master

O git trava e não sai do lugar, quando ele deveria pedir logo em seguida a senha.

Procurando uma solução vamos no github e pegamos o código que tem nele, ensinando a subir os arquivos. O código está com o nome da branch default "main" e após executar o comando na maquina descrito abaixo

git branch -M main
git push -u origin main

e em seguida tentar subir os arquivos

$ git push origin master

apresenta esse erro 

$ git push origin master
error: src refspec master does not match any
error: failed to push some refs to 'https://github.com/andrejs400/livro-receitas.git'

**Reply by fecds** [reply]
Na aula ao criar o repositório no github ele diz para não marcar Add a README file, pois o arquivo foi criado na maquina local e será um dos documentos enviado para o github,

Ele não manda marcar a configuração Add .gitignore e com isso ao criar o repositório, você não consegue mudar a branch dele.

Ao quebrar a cabeça com um amigo vimos que ao marcar na criação do repositório a configuração Add .gitignore e em seguida também já escolhemos o template do java, o repositório é criado com um outro layout e ai conseguimos mudar a branch desse novo repositório de MAIN para MASTER.

**Reply by fabiotbraga** [reply]
To com o mesmo problema, não consigo upar esse repositório no github, já fiz o commit, mudei a branch do github pra "master" e nada funcionou

**Reply by ojvribeiro** [reply]
@fabiotbraga você pode colocar o conteúdo do seu terminal aqui pra eu saber o que aconteceu no seu caso?

**Reply by fabiotbraga** [reply]
Eu consegui resolver, os meus comandos eram parecidos com os que postaram aqui, mudando só os nomes de arquivo, pasta e afins
Era algum problema envolvendo a branch mesmo. Procurando ajuda eu consegui fazer funcionar utilizando o comando:
git config --global credential.provider generic
Dps fiz o passo a passo normal:
git branch -m main   
git commit -m "..."	
git push -u origin main
E funcionou, a explicação eu não vou saber exatamente, mas o importante é que rodou kkkk
Se vc souber ou alguém souber seria interessante pra tirar a dúvida mais a fundo.

**Reply by fecds** [reply]
Fabio, após utilizar o mesmo comando "git config --global credential.provider generic" meu git solicitou a senha de conexão, eu só tive que renomear minha Branch local de MASTER para MAIN e depois consegui sincronizar sem problemas.

Obrigado por compartilhar sua solução.

**Reply by evandirsantarita** [reply]
Obrigado a todos e em especial ao fabiotbraga, porque só conseguir também resolver desse jeito:
git config --global credential.provider generic
Dps fiz o passo a passo normal:
git branch -m main
git commit -m "..."
git push -u origin main

**Comment by mrd-silva** [comment]
Boa noite, pessoal.

Também estou fazendo o curso, porém me deparei com outro erro:
`Superuser@DESKTOP-RP3U5I5 MINGW64 /c/workspace/livre-receitas (main)`
`$ git push -u origin main`
`git@github.com: Permission denied (publickey).`
`fatal: Could not read from remote repository.`


Abaixo a lista de comandos que utilizei num ultimo checkpoint:


`Superuser@DESKTOP-RP3U5I5 MINGW64 /c/workspace/livre-receitas (main)`
`$ git status`
`On branch main`
`nothing to commit, working tree clean`

`Superuser@DESKTOP-RP3U5I5 MINGW64 /c/workspace/livre-receitas (main)`
`$ git push -u origin main`
`git@github.com: Permission denied (publickey). `
`fatal: Could not read from remote repository.`

`Please make sure you have the correct access rights`
`and the repository exists.`

`Superuser@DESKTOP-RP3U5I5 MINGW64 /c/workspace/livre-receitas (main)`
`$ git remote add origin git@github.com:mrd-silva/livro-receitas.git`
`error: remote origin already exists.`

`Superuser@DESKTOP-RP3U5I5 MINGW64 /c/workspace/livre-receitas (main)`
`$ git remote -v`
`origin  git@github.com:mrd-silva/livro-receitas.git (fetch)`
`origin  git@github.com:mrd-silva/livro-receitas.git (push)`

Saberiam me ajudar?

**Reply by bigauke** [reply]
ao invés de inserir o nome de usuário tenta fazer a conexão com seu email e senha nornal.

**Reply by ninetails** [reply]
### Método mais correto e explicativo

Esse erro acontece pois você não configurou as chaves de acesso no teu GitHub. Você provavelmente vai precisar ter instalado OpenSSH (no Google dá pra achar como instalar pro Linux bem fácil), gerar a chave e depois copiar a chave pública pras tuas configurações no GitHub.

Tem um guia certinho na seguinte URL:
https://docs.github.com/pt/github/authenticating-to-github/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account

### Método mais fácil e não tão correto, mas alternativo

Mude seu apontamento de branch da `origin` de utilizar SSH para HTTPS:

https://docs.github.com/pt/github/getting-started-with-github/getting-started-with-git/managing-remote-repositories

Com o guia acima, mude a url do origin que hoje tá como

> git@github.com:mrd-silva/livro-receitas.git

para a de HTTPS

![onde pegar](https://user-images.githubusercontent.com/40065/125693335-d12860b5-d3d6-4b50-9059-8403cab0bd24.png)

> https://github.com/mrd-silva/livro-receitas.git

rodando

> git remote set-url origin https://github.com/mrd-silva/livro-receitas.git

Desta forma, todas as vezes que vc precisar logar no GitHub para checar suas credenciais ele vai pedir as credenciais direto no teu terminal (mas pode ser um saco ter que logar quase toda hora rs)

**Comment by mrd-silva** [comment]
Pessoal, boa noite.

Acabei conseguindo resolver o problema recriando o repositório.

Agradeço pelo suporte!

Abs

Em qua., 14 de jul. de 2021 às 18:14, Carlos Kazuo ***@***.***>
escreveu:

> Método mais fácil
>
> Mude seu apontamento de branch da origin de utilizar SSH para HTTPS:
>
>
> https://docs.github.com/pt/github/getting-started-with-github/getting-started-with-git/managing-remote-repositories
>
> Com o guia acima, mude a url do origin que hoje tá como
>
> ***@***.***:mrd-silva/livro-receitas.git
>
> para a de HTTPS
>
> [image: onde pegar]
> <https://user-images.githubusercontent.com/40065/125693335-d12860b5-d3d6-4b50-9059-8403cab0bd24.png>
>
> https://github.com/mrd-silva/livro-receitas.git
>
> rodando
>
> git remote set-url origin https://github.com/mrd-silva/livro-receitas.git
>
> Desta forma, todas as vezes que vc precisar logar no GitHub para checar
> suas credenciais ele vai pedir as credenciais direto no teu terminal (mas
> pode ser um saco ter que logar quase toda hora rs)
> Método mais correto e explicativo
>
> Esse erro acontece pois você não configurou as chaves de acesso no teu
> GitHub. Você provavelmente vai precisar ter instalado OpenSSH (no Google dá
> pra achar como instalar pro Linux bem fácil), gerar a chave e depois copiar
> a chave pública pras tuas configurações no GitHub.
>
> Tem um guia certinho na seguinte URL:
>
> https://docs.github.com/pt/github/authenticating-to-github/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/frontendbr/forum/discussions/1924#discussioncomment-1005911>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATAGBOVC4QDPMUU5KBJKPCTTXX44VANCNFSM47M5R5CQ>
> .
>


-- 
*Marcos Silva*
*Mobile*: *+55 11 98764-4774*

**Comment by helen-knk** [comment]
Falaaa Galera, tudo bem com vcs?

Também estava um problema parecido:
```
$ git push -u origin main
To https://github.com/goetah/initial-login-screen
 ! [rejected]        main -> main (fetch first)
error: failed to push some refs to 'https://github.com/goetah/initial-login-screen'
```
Fiz os passos:
- [X] git status
- [X] git add
- [X] commit inicial
- [X] push para o repositório

e ainda aparecia o erro!

Como já foi mencionado na discussão, acontece que o git inicia no branch Master, porém o GitHub no Main ~tururu

Para arrumar isso, alterei o git para main e forcei o push:
```
$ git branch -m master main
$ git push origin main --force
```

ou assim:
```
$ git branch -m master main
$ git push origin main -f
```

Espero ter ajudado de alguma forma, abraços e até a próxima :D
🚀🚀🚀

**Comment by FelipeMartinsFrudeli** [comment]
tenta o "git push -f origin main" ou "git push -f origin master"

**Comment by AnaBerigo** [comment]
Eu descobri qual foi o meu erro, não sei se é o caso.

Segui as instruções que foram ditas anteriormente e nada funcionou até que li a explicação dessa pessoa
 https://pt.stackoverflow.com/questions/21243/problema-pra-publicar-projeto-github

Refiz meu projeto e fiz esses passos:

```
anabe@ACER-ANA MINGW64 ~/Desktop/TesteGit/Teste2
$ git init
anabe@ACER-ANA MINGW64 ~/Desktop/TesteGit/Teste2 (master)
$ git status
anabe@ACER-ANA MINGW64 ~/Desktop/TesteGit/Teste2 (master)
$ git branch -M main
anabe@ACER-ANA MINGW64 ~/Desktop/TesteGit/Teste2 (main)
$ git add .
anabe@ACER-ANA MINGW64 ~/Desktop/TesteGit/Teste2 (main)
$ git commit -m "Meu primeiro commit"
anabe@ACER-ANA MINGW64 ~/Desktop/TesteGit/Teste2 (main)
$ git remote add origin git@github.com:AnaBerigo/Teste.git
anabe@ACER-ANA MINGW64 ~/Desktop/TesteGit/Teste2 (main)
$ git push -u origin main
Enter passphrase for key '/c/Users/anabe/.ssh/id_ed25519':
```
No início eu não entendia o erro, mas compreendi que tem que trocar para main antes de usar o comando add. Adicionei também a chave SSH que o @ninetails sugeriu, o que pode ter influenciado na segunda tentativa.
Espero ajudar. Abraços!

**Comment by AnaBerigo** [comment]
O que houve?

Em ter., 21 de set. de 2021 às 22:00, Ezequiel Alves <
***@***.***> escreveu:

**Comment by Martinhadoba** [comment]
Boa tarde, estou tentando Gerar os pares de chave SSH no meu Git Bash mas está retornando a seguinte mensagem:

Saving key "/p//.ssh/id_ed25519" failed: No such file or directory


Vocês podem me ajudar? Rs Obrigada!

**Reply by jampow** [reply]
Parece que está tentando salvar numa pasta inválida `/p//.ssh/`.

Se estiver com este problema ainda, manda o comando que está dando para gerar a chave e o sistema operacional que está usando

**Comment by LauraBarbosa564** [comment]
Boa tarde!

Alguém já teve esse problema?

error: failed to push some refs to 'https://github.com/LauraBarbosa564/cookbook.
git'

**Reply by ghost** [reply]
Oie, tenta rodar antes de subir:

```
git pull --rebase
```

Tenta ver se teu repo tá sincronizado com o `origin`.

**Reply by bpires** [reply]
Provavelmente é como a colega falou acima, você está tentando fazer um push para uma branch remota que está à frente da sua branch de trabalho local (tem novos commits). 

Antes de dar push, portanto, é necessário que você faça um `git pull origin` para receber as últimas alterações feitas na versão da origin. 

Se foi você mesma quem fez as alterações na origin e quiser sobrescrevê-las, você ainda pode usar a opção `--force` na hora de fazer o push à força, porém o histórico de commits do remote será sobrescrito! ⚠️
